app/models/manageiq/providers/openstack/infra_manager/host.rb

```diff
@@ -455,4 +455,31 @@ def refresh_network_interfaces(ssu)
   def self.display_name(number = 1)
     n_('Host (OpenStack)', 'Hosts (OpenStack)', number)
   end
+
+  def self.post_refresh_ems(ems_id, _update_start_time)
+    ems = ExtManagementSystem.find(ems_id)
+    hosts = ems.hosts
```

Comment:
You should be able to limit this down to hosts created in the last refresh e.g. https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_or_template.rb#L1106

Reply:
Is it going to query hosts if first refresh fails?

Reply:
`post_refresh` is only called on successful refresh https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/base_manager/refresher.rb#L62-L63